pkg/asset/installconfig/powervs/validation_test.go

```diff
@@ -126,3 +129,85 @@ func TestValidate(t *testing.T) {
 		})
 	}
 }
+
+func TestValidatePreExistingPublicDNS(t *testing.T) {
+	cases := []struct {
+		name     string
+		edits    editFunctions
+		errorMsg string
+	}{
+		{
+			name:     "no pre-existing DNS records",
+			errorMsg: "",
+		},
+		{
+			name:     "pre-existing DNS records",
+			errorMsg: `^\[baseDomain\: Duplicate value\: \"record api\.valid-cluster-name\.valid\.base\.domain already exists in CIS zone \(valid-zone-id\) and might be in use by another cluster, please remove it to continue\", baseDomain\: Duplicate value\: \"record api-int\.valid-cluster-name\.valid\.base\.domain already exists in CIS zone \(valid-zone-id\) and might be in use by another cluster, please remove it to continue\"\]$`,
+		},
+		{
+			name:     "cannot get zone ID",
+			errorMsg: `^baseDomain: Internal error$`,
+		},
+		{
+			name:     "cannot get DNS records",
+			errorMsg: `^baseDomain: Internal error$`,
+		},
+	}
+	setMockEnvVars()
+
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+
+	powervsClient := mock.NewMockAPI(mockCtrl)
+	metadata := mock.NewMockMetadataAPI(mockCtrl)
+
+	dnsRecordNames := [...]string{fmt.Sprintf("api.%s.%s", validClusterName, validBaseDomain), fmt.Sprintf("api-int.%s.%s", validClusterName, validBaseDomain)}
+
+	//metadata := powervs.NewMetadata(validBaseDomain)
+	//metadata.SetCISInstanceCRN(validCISInstanceCRN)
+
+	// Mocks: no pre-existing DNS records
+	metadata.EXPECT().CISInstanceCRN(gomock.Any()).Return(validCISInstanceCRN, nil)
```

Comment:
instead of adding the same mock results you can change 
`metadata.EXPECT().CISInstanceCRN(gomock.Any()).Return(validCISInstanceCRN, nil)`
to 
`metadata.EXPECT().CISInstanceCRN(gomock.Any()).Return(validCISInstanceCRN, nil).AnyTimes()`
since it is always returning the same result.

Reply:
same applies for your mock call to `GetDNSZoneIDByName` call

Reply:
This worked for `CISInstanceCRN` but `GetDNSZoneIDByName` expects an empty string for the `cannot get zone ID` test unlike the other tests.